src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -231,6 +233,7 @@ static Integer parseSupComVersion(byte[] rawReplayBytes) {
   static String parseMapName(byte[] rawReplayBytes) {
     int mapDelimiterIndex = Bytes.indexOf(rawReplayBytes, new byte[]{0x00, 0x0D, 0x0A, 0x1A});
     String mapPath = new String(rawReplayBytes, MAP_NAME_OFFSET, mapDelimiterIndex - MAP_NAME_OFFSET, US_ASCII);
+    // TODO make sure that these are valid characters for a file name? this is potentially unsafe user input
```

Comment:
Can u specify what you mean, maybe link a GitHub issue. Donot think that a user could use this string to attack another client

Reply:
The usual maps dont require calling this beause mapName is in metadata. That mapName is retrived from some server column by matching via filename. So for normal maps we are guarantueed to get server entries. When directly parsing, we are reading a name that comes directly from one players client.

I guess it doesnt matter because downloading the mapo will simply fail if we have invalid characters right?

Reply:
I added in a check for invalid filename characters